x-pack/plugins/security_solution/common/api/timeline/delete_note/delete_note_route.gen.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/*
+ * NOTICE: Do not edit this file manually.
+ * This file is automatically generated by the OpenAPI Generator, @kbn/openapi-generator.
+ *
+ * info:
+ *   title: Elastic Security - Timeline - Notes API
+ *   version: 2023-10-31
+ */
+
+import { z } from 'zod';
+
+export type DeleteNoteRequestBody = z.infer<typeof DeleteNoteRequestBody>;
+export const DeleteNoteRequestBody = z.union([
```

Comment:
@janmonschke, I'm not sure if you made the changes for this one, but I think you may have made some of these fields required, which I agree with generally, but this will change the current io-ts expectation which I think is for this

```
{
  noteId?: string | null;
  noteIds?: string[] | null;
 }
```

I would make the change for `noteIds` at least to be z.array(z.string()).nullable() as we don't want a null item in the array

Reply:
I see, makes sense. Fixed in https://github.com/elastic/kibana/pull/188844/commits/d66b53c9dbcb4c47752b981da9606ba9fc3fe463